### PR TITLE
gRPC scaffolding, including REST translator

### PR DESCRIPTION
**Pull request by EnergySRE**
Includes a few different pieces:
- Python gRPC server (:8191) with a placeholder implementation.
- Go gateway (:8190), translates REST to gRPC following recipe in proto file.
- Pipenv and Go module setup.

### CODE REVIEW

File: orchestrator/Makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+include ../Makefile
+
+GOOGLEAPIS=github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis
```
Comment (aaj-gd) [inline orchestrator/Makefile]: Im not sure if this is the best place to put it but i had to git clone and move those files into that location.  i dont know if theres a magic go get command that would do that.

File: orchestrator/Makefile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+api.pb.go:
+	protoc -I. \
+		--proto_path=${GOPATH}/src \
+		--proto_path=${GOPATH}/src/${GOOGLEAPIS} \
+		--go_out=plugins=grpc:. \
+		--plugin=${GOBIN}/protoc-gen-go \
+		api.proto
+
+# HTTP REST to gRPC adapter
+api.pb.gw.go:
```
Comment (aaj-gd) [inline orchestrator/Makefile]: minor nit for me is why not output to an `autogen/` directory i know you .gitgnore the generated protobuf but this way you can just ignore the directory and makes the directory slightly cleaner

File: .gitignore [hunk 3]
```diff
@@ -227,4 +227,11 @@ tags
...
 # END AUTOGEN: Put manual additions below, if not available via gitignore.io. ==
-# ==============================================================================
\ No newline at end of file
+# ==============================================================================
+
+# Local GOPATH
+go/
+
+# Protocol Buffers generated code
+*.pb*.go
```
Comment (aaj-gd) [inline .gitignore]: minor nit. can we just put all the protobufs in an `autogen/` directory so you can just ignore that directory.  it also makes the directory structure cleaner.

File: Pipfile [hunk 4]
```diff
@@ -0,0 +1,17 @@
+[[source]]
```
Comment (aaj-gd) [inline Pipfile]: either here as a comment or in the readme.
```
pip install pipenv
pipenv install
```

File: orchestrator/Makefile (hunk 1 above)
Comment (aaj-gd) [inline orchestrator/Makefile]: I have no idea what the magic go command is to get that directory in the gopath.  i ended up just git cloning and then moving the files over. Is there a better way to do that?

File: orchestrator/server.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                         interceptors=[_LoggingInterceptor()])
+    o = Orchestrator()
+    svc.add_OrchestratorServicer_to_server(o, server)
+
+    reflectable = [
+        pb.DESCRIPTOR.services_by_name['Orchestrator'].full_name,
+        reflection.SERVICE_NAME,
+    ]
+    reflection.enable_server_reflection(reflectable, server)
+
```
Comment (aaj-gd) [inline orchestrator/server.py]: can you add a log line to say listeing on port XXX, when i run this, there's no output so im not sure if its working

File: orchestrator/proxy/proxy.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  ctx, cancel := context.WithCancel(ctx)
+  defer cancel()
+
+  mux := runtime.NewServeMux()
+  opts := []grpc.DialOption{grpc.WithInsecure()}
+  err := gw.RegisterOrchestratorHandlerFromEndpoint(ctx, mux,  *backendFlag, opts)
+  if err != nil {
+    return err
+  }
+
```
Comment (aaj-gd) [inline orchestrator/proxy/proxy.go]: add a log line of listening on port XXX

File: orchestrator/Makefile (hunk 1 above)
Comment (aaj-gd) [reply orchestrator/Makefile]: just verified codys new commit works. so resolving

File: .gitignore (hunk 3 above)
Comment (EnergySRE) [reply .gitignore]: Not philosophically opposed to that, but Go packages are sensitive to directory structure, so it'd just move the icky out of gitignore and into import statements for the package clients. Although if it's standard practice, I'd be inclined to conform. What do you think?

Comment (aaj-gd) [reply .gitignore]: that works too.  even a readme or comment line somewhere saying these are generated by something.

Comment (EnergySRE) [reply .gitignore]: Hmmm, I guess I don't see what's wrong with the existing comment on line 235. It says, "Protocol Buffers generated code".  Where else do you think one would stumble on this?

File: Pipfile (hunk 4 above)
Comment (EnergySRE) [reply Pipfile]: Good idea, thanks. I added that to `make env`, and added a couple of how-to comments here.

File: orchestrator/proxy/proxy.go (hunk 6 above)
Comment (EnergySRE) [reply orchestrator/proxy/proxy.go]: I0320 10:13:36.473587   38532 proxy.go:31] Listening on :8190

File: orchestrator/server.py (hunk 5 above)
Comment (EnergySRE) [reply orchestrator/server.py]: INFO:root:Listening on [::]:8191

**Comment by EnergySRE** [comment]
Submitting now, based on your earlier approval. Please just let me know if you want me to add comments elsewhere.

File: .gitignore (hunk 3 above)
Comment (aaj-gd) [reply .gitignore]: youd have to look at the .gitignore.  i guess for those familiar with protobufs its not a big deal its just we should say somewhere possibly in the README file some files are autogenerated.

**Comment by aaj-gd** [comment]
everything looks good

**Pull request merged**